### ISSUE TITLE
FutureTest fails (undeterministic?)

### ISSUE THREAD

**Issue by danieldietrich** [javaslang/javaslang#438]
I commented the whole test class in order to temporary fix the issue.

To be fixed:

```java
Failed tests:   basicMapTest(javaslang.concurrent.FutureTest): Checker was supposed to be empty, but was not.  expected:<0> but was:<1>
```

**Comment by DillonJettCallis** [comment]
Sorry, that's because future.await continues after call backs have been TRIGGERED but not after they've finished. I should have known that when I wrote the tests.

**Comment by danieldietrich** [comment]
Hi Dillon,
ok. I'm on vacation this week - could you uncomment the tests, please, and fix it?
After finishing the collections I will spend more time on the futures. E.g. the FilterMonadic interface may be implemented.
Greets,
Daniel

**Comment by danieldietrich** [comment]
Fixed with #633 by Dillon
